File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        //if (nums.length == 1) return 0;
        int cnt = 0;
        for (int i = nums.length - 1; i > 0; ){
            boolean flag = false;
            int temp = 0;
            for (int j = i - 1; j >= 0; j--){
                if (i - j <= nums[j]){
                    temp = j;
                    flag = true;
                }
            }
            if(flag) {
                cnt++;
                i = temp;
            }
            else i--;
        }
        return cnt;
    }
}

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int far = nums[0];
        int cur = 0;
        int jump = 0;
        for (int i = 0; i < n - 1; i++){
            far = max(far, i + nums[i]);
            if (i == cur){
                jump++;
                cur = far;
            }
        }
        return jump;
    }
};